plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/rule/storage/TrieBasedStore.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.rule.storage;
+
+import org.apache.commons.collections4.trie.PatriciaTrie;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * This is a patricia trie based implementation of AttributeValueStore
+ */
+public class TrieBasedStore implements AttributeValueStore {
+    PatriciaTrie<String> trie;
+
+    public TrieBasedStore(PatriciaTrie<String> trie) {
+        this.trie = trie;
+    }
+
+    @Override
+    public void addValue(String key, String value) {
+        trie.put(key, value);
+    }
+
+    @Override
+    public void removeValue(String key) {
+        trie.remove(key);
+    }
+
+    @Override
+    public Optional<String> getValue(String key) {
+        /**
+         * Since we are inserting prefixes into the trie and searching for larger strings
```

Comment:
I think adding more context/doc here(with a simple example) or above the class would be helpful. Mostly how PatriciaTrie works on a high level ie it stores compressed paths rather than individual characters, uses bits etc, and how binary search helps here etc. 

Makes it easy for the reader to gain enough context to go through this.

Reply:
That is true, I can add some examples but adding the opensource data structure structure mechanism details would be redundant here since the details are available online through various resources such as apache commons javadoc etc.

Reply:
Mostly adding info as to why we chose PatriciaTrie instead of using a regular/other trie should be enough. As it will include enough details.